slices/libpam-runtime.yaml

```diff
@@ -18,28 +18,125 @@ slices:
+          elif vals[0] in ["Auth-Type", "Account-Type", "Session-Type", "Password-Type"]:
+            m, t = parse_type(vals)
+          elif vals[0] == "Session-Interactive-Only":
+            if vals[1].strip() == "yes":
+              m = "session"
+          elif vals[0] in ["Auth", "Account", "Session", "Password"]:
+            d = []
+            for j in range(i + 1, len(lines)):
+              if ":" in lines[j]:
+                break
+              d.append(lines[j])
+            confdata[m] = {t: {p: d}}
+
+      def reconfigure_data_line(section, line, i):
+        res = ""
+        upd = line.replace("success=end", "success=" + str(i))
+        if section == "session-noninteractive":
+          res += "session" + upd + "\n"
+        else:
+          res += section + upd + "\n"
+        return res
+
+      def build_block(data, section, block, existing):
+        si = 1
+        res = existing
+        if section in data:
+          types = data[section]
+          if block in types:
+            # get the keys (which are the priorities) and
+            # then reverse sort them to get the highest priority
+            # first
+            itemsByPriority = types[block]
+            keys = itemsByPriority.keys()
+            keys = sorted(keys, reverse=True)
+            for p in keys:
+              for d in itemsByPriority[p]:
+                res = reconfigure_data_line(section, d, si) + "\n"
+                si += 1
+
+        # no primary block, so output a stock pam_permit line
+        # to keep the stack intact
+        if res == "" and block == "Primary":
+          return reconfigure_data_line(section, "\t[default=1]\t\t\tpam_permit.so\n", 1);
+        return res
+
+      fnames = ["account", "auth", "password", "session", "session-noninteractive"]
+      idnames = ["$account", "$auth", "$password", "$session", "$session_nonint"]
 
-  # Used by pam-auth-update to generate the pam.d/ config
-  # files inside etc. However pam-auth-update relies on
-  # debconf to actually work and discover these packages.
+      for i in range(len(fnames)):
+        fn = fnames[i]
+        template = content.read("/usr/share/pam/common-" + fn)
+        pb = build_block(confdata, fn, "Primary", "")
+        ab = build_block(confdata, fn, "Additional", "")
+
+        # session also includes settings from the session-noninteractive,
+        # but not the other way around
+        if fn == "session":
+          pb = build_block(confdata, "session-noninteractive", "Primary", pb)
+          ab = build_block(confdata, "session-noninteractive", "Additional", ab)
+
+        template = template.replace(idnames[i] + "_primary", pb)
+        template = template.replace(idnames[i] + "_additional", ab)
+        content.write("/etc/pam.d/common-" + fn, template)
+
+  # the following two slices are kept for backwards-compatibility
+  # however the pam-defaults have been modified to deliver correct
+  # files instead of templates
```

Comment:
```suggestion
  # The following two slices are kept for backwards compatibility.
  # However, the pam-defaults have been modified to deliver correct
  # files instead of templates.
```